### ISSUE TITLE
Fix automatic deployment

### ISSUE THREAD

**Issue by ulope** [raiden-network/raiden#1893]
## Problem Definition

During the last release we noticed that the automated release system doesn't work correctly.
Needs to be fixed.

Details in the [travis build](https://travis-ci.org/raiden-network/raiden/builds/405474632)

**Comment by palango** [comment]
Can we close this now?

**Comment by ulope** [comment]
@palango Yep, it something still breaks we'll make a new one

Fixed with #1952 and #1985

**Comment by LefterisJP** [comment]
@ulope Seems it's still only the OSX build that was botched.

**Comment by LefterisJP** [comment]
From our [build](https://travis-ci.org/raiden-network/raiden/builds/414675294) for v0.5.0 both automatic deployment jobs failed.

- The [linux](https://travis-ci.org/raiden-network/raiden/jobs/414675310) job made it very far. Enough to make the binary and give it to Github. But failed at the automatic Pypi upload:

```
HTTPError: 400 Client Error: User 'raiden_pypi_automated' does not have a verified primary email address. Please add a verified primary email before attempting to upload to PyPI. See https://pypi.org/help/#verified-email for more information.for more information. for url: https://upload.pypi.org/legacy/
```

- The [OSX](https://travis-ci.org/raiden-network/raiden/jobs/414675311) Job is unfortunately still failing.

**Comment by ulope** [comment]
Both should be fixed with #2204.

I've temporarily disabled the automatic PyPI upload and created an issue to track re-enabling it: #2205

**Comment by LefterisJP** [comment]
I am reopening this **again**, as OSX deployment did not work in the latest release. https://travis-ci.org/raiden-network/raiden/builds/423123306

Here is the actual OSX deploy job:
https://travis-ci.org/raiden-network/raiden/jobs/423123329

It succeeds but no binary is generated. The only relevant output is:

```
Skipping a deployment with the releases provider because this branch is not permitted: v0.7.0
Skipping a deployment with the script provider because this branch is not permitted: v0.7.0
```
